tests/test_cacerts_from_env.py

```diff
@@ -0,0 +1,57 @@
+import os
+import sys
+import mock
+import httplib2
+
+CA_CERTS_BUILTIN = os.path.join(os.path.dirname(httplib2.__file__), "cacerts.txt")
+ENV_CERTS_FILE = "unittest_env_certs_file"
+CERTIFI_CERTS_FILE = "unittest_certifi_file"
+CUSTOM_CA_CERTS = "unittest_custom_ca_certs"
+
+
+def test_certs_file_from_builtin():
+    httplib2.certs.certifi_available = False
```

Comment:
This assignment affects other tests, so should be returned to previous value.

Something like this will reliably return global state.
```
with mock.patch("httplib2.certs.certifi_available", True):
  assert where()...
```

Reply:
Done. I used @mock.patch for the available flags

Should I use them for the where methods as well? Then I'll need to define them as None before try except block so that they are defined and I can patch them.

Reply:
Yes, it's not about any particular flag. Rule is: tests must either pass or fail only because main code is valid or not.

Tests must behave same way ...
- when certifi/custom_ca_loader is installed and not, unless test suite installs package in controllable manner
- when some environment variable is set or not, unless test runner controls environment
- when tests order were shuffled - this excludes modification of any global variables

Reply:
Please have patience mate, I'm sorry this is not ideal one step review like change in these places then merged. You're doing important work.